draft-tuexen-tsvwg-rfc4960-errata.xml

```diff
@@ -733,6 +733,74 @@ partial_bytes_acked and cwnd in the congestion avoidance phase.</t>
 </section>
 </section>
 
+<section title='HEARTBEAT ACK and the association error counter'>
+<section title='Description of the Problem'>
+<t>Section 8.1 and Section 8.3 of <xref target="RFC4960"/>
+prescribe that the receiver of a HEARTBEAT ACK must reset the
+association overall error counter. In some circumstances, e.g. when
+a router discards DATA chunks but not HEARTBEAT chunks due to the
+larger size of the DATA chunk, it is better to not clear the
+association error counter by the HEARTBEAT ACK and reset it by the
+SACK only.</t>
+</section>
+<section title='Text Changes to the Document'>
+<figure>
+<artwork>
+---------
+Old text: (Section 8.1)
+---------
+
+The counter shall be reset each time a DATA chunk sent to that peer
+endpoint is acknowledged (by the reception of a SACK) or a HEARTBEAT
+ACK is received from the peer endpoint.
+
+---------
+New text: (Section 8.1)
+---------
+
+The counter shall be reset each time a DATA chunk sent to that peer
+endpoint is acknowledged (by the reception of a SACK). When a
+HEARTBEAT ACK is received from the peer endpoint, the counter should
+also be reset. The receiver of the HEARTBEAT ACK may choose not to
+clear the counter if there is outstanding data on the association.
+This allows for handling the possible difference in reachability
+based on DATA chunks and HEARTBEAT chunks. 
+
+---------
+Old text: (Section 8.3)
+---------
+
+Upon the receipt of the HEARTBEAT ACK, the sender of the HEARTBEAT
+should clear the error counter of the destination transport address
+to which the HEARTBEAT was sent, and mark the destination transport
+address as active if it is not so marked.  The endpoint may
+optionally report to the upper layer when an inactive destination
+address is marked as active due to the reception of the latest
+HEARTBEAT ACK.  The receiver of the HEARTBEAT ACK must also clear the
+association overall error count as well (as defined in Section 8.1).
+
+---------
+New text: (Section 8.3)
+---------
+
+Upon the receipt of the HEARTBEAT ACK, the sender of the HEARTBEAT
+should clear the error counter of the destination transport address
+to which the HEARTBEAT was sent, and mark the destination transport
+address as active if it is not so marked. The endpoint may
+optionally report to the upper layer when an inactive destination
+address is marked as active due to the reception of the latest
+HEARTBEAT ACK. The receiver of the HEARTBEAT ACK should also clear
+the association overall error counter (as defined in Section 8.1).
+</artwork>
```

Comment:
"must" is changed to "should" in the last sentence. I've also removed "as well" as it seems to be excessive.